File: Palindromo.java
import java.util.Scanner;

public class Palindromo{

 	public static void main(String[] args){
 		
       	String palabra="";
 		String palabra_invertida="";
 		int x;

        System.out.println("\t\t\n\nPROGRAMA QUE VERIFICA SI UNA CADENA ES UN PALINDROMO\n\n\n");
        System.out.println("IMPORTANTE: La cadena debe ir sin espacios \n\n");
        System.out.print("Escribe una cadena para verificar si es un palindromo:");
        Scanner scan = new Scanner(System.in);
        palabra= scan.nextLine();             
 
        for (x = palabra.length()-1; x>=0; x--){
       		 palabra_invertida = palabra_invertida + palabra.charAt(x);
       	}
			 if  (palabra.equalsIgnoreCase (palabra_invertida)){
				 System.out.println("Es un palindromo");
			 }
			 else{
			 System.out.println("No es un palindromo");
			 } 
		
	}
}

File: palindromo.c
#include <stdio.h>
#include <string.h>

int main()
{
	int palindromo = 1;
	int i, j;
	char palabra[40];

	printf("\t\tPROGRAMA QUE VERIFICA SI UNA CADENA ES UN PALINDROMO\n\n\n");
	printf("IMPORTANTE: La cadena debe ir sin espacios \n\n");
	printf("Escribe una cadena para verificar si es un palindromo: ");
	fflush(stdout);
	gets(palabra);
	j = strlen(palabra) - 1;
	for (i = 0; i<strlen(palabra) / 2; i++, j--) {
			if (*(palabra + i) != *(palabra + j)) {
			palindromo = 0;
			break;
		}
	}
	if (palindromo)
		printf("\n Es un palindromo");
	else
		printf("\n No es un palindromo");
	getchar();
	return (0);
}
